engine/flink/kafka-util/src/main/scala/pl/touk/nussknacker/engine/kafka/consumerrecord/ConsumerRecordToJsonFormatter.scala

```diff
@@ -0,0 +1,87 @@
+package pl.touk.nussknacker.engine.kafka.consumerrecord
+
+import java.nio.charset.StandardCharsets
+
+import com.github.ghik.silencer.silent
+import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}
+import io.circe.{Decoder, Encoder}
+import org.apache.flink.streaming.connectors.kafka.{KafkaDeserializationSchema, KafkaSerializationSchema}
+import org.apache.kafka.clients.consumer.ConsumerRecord
+import org.apache.kafka.common.header.Headers
+import org.apache.kafka.common.record.TimestampType
+import pl.touk.nussknacker.engine.api.CirceUtil
+import pl.touk.nussknacker.engine.api.test.{TestDataSplit, TestParsingUtils}
+import pl.touk.nussknacker.engine.kafka.{ConsumerRecordUtils, RecordFormatter}
+import pl.touk.nussknacker.engine.kafka.consumerrecord.SerializableConsumerRecord._
+
+import scala.annotation.nowarn
+
+@silent("deprecated")
+@nowarn("cat=deprecation")
+class ConsumerRecordToJsonFormatter[K: Encoder:Decoder, V: Encoder:Decoder](deserializationSchema: KafkaDeserializationSchema[ConsumerRecord[K, V]],
+                                                                            serializationSchema: KafkaSerializationSchema[ConsumerRecord[K, V]])
+  extends RecordFormatter {
+
+  implicit val consumerRecordDecoder: Decoder[SerializableConsumerRecord[K, V]] = deriveDecoder
+  implicit val consumerRecordEncoder: Encoder[SerializableConsumerRecord[K, V]] = deriveEncoder
+
+  override protected def formatRecord(record: ConsumerRecord[Array[Byte], Array[Byte]]): Array[Byte] = {
+    val deserializedRecord = deserializationSchema.deserialize(record)
+    val serializableRecord = SerializableConsumerRecord(
+      Option(deserializedRecord.key()),
+      deserializedRecord.value(),
+      Option(deserializedRecord.topic()),
+      Option(deserializedRecord.partition()),
+      Option(deserializedRecord.offset()),
+      Option(deserializedRecord.timestamp()),
+      Option(ConsumerRecordUtils.toMap(deserializedRecord.headers()).mapValues(s => Option(s)))
+    )
+    implicitly[Encoder[SerializableConsumerRecord[K, V]]].apply(serializableRecord).noSpaces.getBytes(StandardCharsets.UTF_8)
```

Comment:
Why you explicitly derive encoder, make it implicit and after that implicitly derive it again? After making it implicit, you can use just `serializableRecord.asJson` or if you don't like implicit extensions (Ops syntax) you can remove implicit from `consumerRecordEncoder` declaration and use just `consumerRecordEncoder(serializableRecord)`